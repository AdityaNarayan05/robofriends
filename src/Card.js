import React from 'react';


const Card=()=>{
    return(
        <div>
            <img alt='robots' src ='https://robohash.org/mala?200*200' />
            <div>
                <h2> Aditya Narayan Jaiswal </h2>
                <p>nlaxmi886@gmail.com</p>
            </div>
        </div>
    );
}

export default Card;