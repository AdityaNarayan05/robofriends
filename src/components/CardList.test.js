import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';
import Card from './Card';

describe('CardList', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Doe',
      email: 'johndoe@example.com',
    },
    {
      id: 2,
      name: 'Jane Smith',
      email: 'janesmith@example.com',
    },
  ];

  it('should render CardList component', () => {
    const wrapper = shallow(<CardList robots={mockRobots} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct number of Card components', () => {
    const wrapper = shallow(<CardList robots={mockRobots} />);
    expect(wrapper.find(Card)).toHaveLength(mockRobots.length);
  });

  it('should pass correct props to Card components', () => {
    const wrapper = shallow(<CardList robots={mockRobots} />);
    wrapper.find(Card).forEach((node, index) => {
      expect(node.prop('id')).toEqual(mockRobots[index].id);
      expect(node.prop('name')).toEqual(mockRobots[index].name);
      expect(node.prop('email')).toEqual(mockRobots[index].email);
    });
  });
});
