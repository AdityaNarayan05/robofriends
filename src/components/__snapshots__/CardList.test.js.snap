// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList should render CardList component 1`] = `
<div>
  <Card
    email="johndoe@example.com"
    id={1}
    key="0"
    name="John Doe"
  />
  <Card
    email="janesmith@example.com"
    id={2}
    key="1"
    name="Jane Smith"
  />
</div>
`;
